src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
+// import { threadId } from "worker_threads";
+
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('your error message');
+    }
+
+    const arrayOfSides = [this.a, this.b, this.c];
+    /* eslint-disable */
+    const longest = arrayOfSides.reduce((prev, side) => (prev < side ? side : prev), 0);
+
+    const index = arrayOfSides.indexOf(longest);
+
+    const sumOfTwoSides = arrayOfSides.filter((side) => (
+      side !== arrayOfSides[index]))
+      .reduce((prev, side) => prev + side, 0);
 
+    if (arrayOfSides[index] >= sumOfTwoSides) {
+      throw new Error('your error message');
+    }
+  }
```

Comment:
That is too much of logic. Just find the biggest side by sorting the array of sides. Then compare it to sum of 2 others sides.